creator/ingest_runs/data_generator/study_generator.py

```diff
@@ -0,0 +1,504 @@
+
+        # Create new or read in existing study data files
+        kf_ids = self.generate_files(create=create_files)
+
+        # Ingest study data files into Data Service
+        self.ingest_pipeline.run()
+
+        # Ensure all kf_ids were loaded
+        for endpoint, kfids in kf_ids.items():
+            self._assert_load(endpoint, kfids)
+
+    def generate_files(self, create=False):
+        """
+        Generate clinical and genomic data files for a test study
+
+        :param create: Create new data files or read existing files
+        :type create: bool
+        """
+        ra.seed(0)
+        self.logger.info(
+            f"Creating clinical and genomic files for {self.study_id}"
+        )
+
+        # Create clinical and genomic data files
+        bio_df = self._create_bio_manifest(create=create)
+        gf_df = self._create_gfs(create=create)
+
+        # Upload unharmonized and harmonized gfs to S3
+        if create:
+            self._upload_gfs_to_s3()
+
+        # Write S3 object manifests for genomic files to disk
+        s3_df = self._create_gf_s3_manifests(create=create)
+
+        # Create BIX GWO Manifest and upload to S3
+        if create:
+            dfs = bio_df, gf_df, s3_df
+        else:
+            dfs = None
+        gwo_df = self._create_bix_gwo_manifest(dfs=dfs)
+
+        # Return kf ids for later validation
+        return {
+            "participants": bio_df["kf_id_participant"].unique(),
+            "biospecimens": bio_df["kf_id_biospecimen"].unique(),
+            "families": bio_df["kf_id_family"].unique(),
+            "genomic-files": gf_df["kf_id_genomic_file"].unique(),
+        }
+
+    def _initialize_study(self):
+        """
+        Create study, sequencing-center in Data Service
+        """
+        self.logger.info(
+            f"Initializing study {self.study_id} in "
+            f"Data Service {DATASERVICE_URL}"
+        )
+        payloads = [
+            {
+                "kf_id": self.study_id,
+                "name": self.study_id,
+                "external_id": self.study_id,
+                "endpoint": "studies"
+            },
+        ]
+        for sc in self.sequencing_centers:
+            payloads.append(
+                {
+                    "kf_id": sc["kf_id"],
+                    "name": sc["name"],
+                    "external_id": sc["name"],
+                    "endpoint": "sequencing-centers"
+                }
+            )
+        for p in payloads:
+            endpoint = p.pop("endpoint", None)
+            url = f"{DATASERVICE_URL}/{endpoint}"
+            self.logger.info(f"Creating {url}")
+            try:
+                resp = requests.post(url, json=p)
+                resp.raise_for_status()
+            except requests.exceptions.HTTPError:
+                ok = resp.status_code == 400 and "already exists" in resp.text
+                if not ok:
+                    raise
+
+    def _kf_id_generator(self, prefix):
```

Comment:
This should probably be imported from `creator/fields.py` oops

Reply:
for some reason, I couldn't get this function to import, so I just copied it over